QuickUnion.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+class QuickUnion {
+
+    constructor() {
+        this._elements = [];
+        this._numberElementsInComponent = [];
+    }
+
+    connect(a, b) {
+        if (this.isConnected(a, b)) return;
+
+        if (isNaN(this._elements[a])) this._elements[a] = a;
+        if (isNaN(this._elements[b])) this._elements[b] = b;
+
+        let rootA = findRoot(a, this._elements);
+        let rootB = findRoot(b, this._elements);
+
+        if (isNaN(this._numberElementsInComponent[rootA])) this._numberElementsInComponent[rootA] = a;
```

Comment:
do you have tests that check this behavior?

the smaler tree should connect to bigger one